manticore/native/models.py

```diff
@@ -140,3 +172,56 @@ def strlen(state, s):
             ret = ITEBV(cpu.address_bit_size, byt == 0, offset, ret)
 
     return ret
+
+
+def strcpy(state: State, dst: Union[int, Expression], src: [int, Expression]) -> int:
+    """
+    strcpy symbolic model
+
+    Algorithm: Copy every byte from the src to dst until finding a byte that will is or must be '\000'
+
+    :param state: current program state
+    :param dst: destination string address
+    :param src: source string address
+    :return: pointer to the dst
+    """
+    if issymbolic(src):
+        raise ConcretizeArgument(state.cpu, 1)
+
+    if issymbolic(dst):
+        raise ConcretizeArgument(state.cpu, 2)
+
+    cpu = state.cpu
+    constrs = state.constraints
+    ret = dst
+    c = cpu.read_int(src, 8)
+    # Copy until '\000' is reached or symbolic memory that can be '\000'
+    while (issymbolic(c) and not Z3Solver.instance().can_be_true(constrs, c == 0)) or c != 0:
+        cpu.write_int(dst, c, 8)
+        src += 1
+        dst += 1
+        c = cpu.read_int(src, 8)
+
+    # Even if the byte was symbolic and constrained to '\000' write a concrete '\000'
+    if (issymbolic(c) and not Z3Solver.instance().can_be_true(constrs, c != 0)) or c == 0:
+        cpu.write_int(dst, 0, 8)
+        return ret
+
+    zeros = _find_zeros(cpu, constrs, src)
+    null = zeros[-1]
+    # If the symmbolic byte was not constrained to '\000' write the appropriate symbolic bytes
+    for offset in range(null, -1, -1):
+        src_val = cpu.read_int(src + offset, 8)
+        dst_val = cpu.read_int(dst + offset, 8)
+        if zeros[-1] == offset:
+            # Make sure last byte of the copy is always a concrete '\000'
+            true_val = ITEBV(cpu.address_bit_size, src_val == 0, 0, src_val)
+            zeros.pop()
+
+        # For every byte that could be null before the current byte add an if then else case to the bitvec tree to set the value to the src or dst byte accordingly
```

Comment:
Nit: This comment is pretty long. `black` doesn't touch comments, but I think we should try to stick to max 100 line column length

Reply:
Will watch for this